packages/core/src/tools/ContactProvider.ts

```diff
@@ -0,0 +1,171 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Platform} from 'react-native';
+import Toast from 'react-native-toast-message';
+import {PERMISSIONS, Permission, request} from 'react-native-permissions';
+import * as Contacts from 'react-native-contacts';
+
+interface ContactData {
+  firstName: string;
+  lastName?: string;
+  phoneNumbers?: {
+    mobilePhone: string;
+    fixedPhone: string;
+  };
+  email?: string;
+  address?: string;
+  notes?: string;
+}
+
+enum PermissionResult {
+  UNAVAILABLE = 'unavailable',
+  BLOCKED = 'blocked',
+  DENIED = 'denied',
+  GRANTED = 'granted',
+  LIMITED = 'limited',
+}
+
+class ContactProvider {
+  constructor() {}
+
+  private _getReadContactPermission = (): Permission => {
+    switch (Platform.OS) {
+      case 'ios':
+        return PERMISSIONS.IOS.CONTACTS;
+      case 'android':
+        return PERMISSIONS.ANDROID.READ_CONTACTS;
+      default:
+        throw new Error('Unsupported device.');
+    }
+  };
+
+  private _showToast = (type: string, message: string): void => {
+    Toast.show({
+      type,
+      position: 'bottom',
+      bottomOffset: 20,
+      text1: type === 'success' ? 'Success' : 'Error',
+      text2: message,
+    });
+  };
+
+  private _requestReadContactsPermission =
+    async (): Promise<PermissionResult> => {
+      try {
+        const reactContactPermission = this._getReadContactPermission();
+        const permissionStatus = await request(reactContactPermission);
+        return permissionStatus as PermissionResult;
+      } catch (error) {
+        console.error('Error requesting contacts permission:', error);
+        return PermissionResult.DENIED;
+      }
+    };
+
+  private _createContactData = (
+    contactData: ContactData,
+  ): Partial<Contacts.Contact> => {
+    const {firstName, lastName, email, phoneNumbers, address, notes} =
+      contactData;
+
+    const contact: Partial<Contacts.Contact> = {
+      givenName: firstName,
+      familyName: lastName || undefined,
+      emailAddresses: email
+        ? [
+            {
+              label: 'email',
+              email,
+            },
+          ]
+        : undefined,
+      phoneNumbers: phoneNumbers
+        ? [
+            {
+              label: 'mobile',
+              number: phoneNumbers.mobilePhone,
+            },
+            {
+              label: 'fixed',
+              number: phoneNumbers.fixedPhone,
+            },
+          ]
+        : undefined,
+      postalAddresses: address
+        ? [
+            {
+              label: 'main address',
+              formattedAddress: address,
+              street: null,
+              pobox: null,
+              neighborhood: null,
+              city: null,
+              region: null,
+              state: null,
+              postCode: null,
+              country: null,
+            },
+          ]
+        : undefined,
+      note: notes || undefined,
+    };
```

Comment:
I still find this construction quite dirty, I'm sure we can do something cleaner for that with a parser/mapper. It will be more maintainable and more flexible if we want to add a new field later. 

The following is working fine but I think we can do something even better by defining type of ContactEntry (phoneNumber, email, address, info) to replace the isArrayField/getData. What do you think ?
```
const CONTACT_MAPPER = {
  firstName: {
    _key: 'givenName',
    getData: (value: string) => value,
    isArrayField: false,
  },
  lastName: {
    _key: 'familyName',
    getData: (value: string) => value,
    isArrayField: false,
  },
  notes: {
    _key: 'note',
    getData: (value: string) => value,
    isArrayField: false,
  },
  mobilePhone: {
    _key: 'phoneNumbers',
    getData: (value: string) => ({
      label: 'mobile',
      number: value,
    }),
    isArrayField: true,
  },
  fixedPhone: {
    _key: 'phoneNumbers',
    getData: (value: string) => ({
      label: 'fixed',
      number: value,
    }),
    isArrayField: true,
  },
  email: {
    _key: 'emailAddresses',
    getData: (value: string) => ({
      label: 'email',
      email: value,
    }),
    isArrayField: true,
  },
  address: {
    _key: 'postalAddresses',
    getData: (value: string) => ({
      label: 'main address',
      formattedAddress: value,
      street: null,
      pobox: null,
      neighborhood: null,
      city: null,
      region: null,
      state: null,
      postCode: null,
      country: null,
    }),
    isArrayField: true,
  },
};

const parseContactData = (_contact: ContactData): Partial<Contacts.Contact> => {
  if (isEmpty(_contact)) {
    return {};
  }

  const result: Partial<Contacts.Contact> = {};

  Object.entries(CONTACT_MAPPER).forEach(([key, config]) => {
    if (_contact[key] != null) {
      const data = config.getData(_contact[key]);

      if (!config.isArrayField) {
        result[config._key] = data;
      } else {
        if (Array.isArray(result[config._key])) {
          result[config._key].add(data);
        } else {
          result[config._key] = [data];
        }
      }
    }
  });

  return result;
};
```